extensions/amp-analytics/0.1/visibility-impl.js

```diff
@@ -508,29 +545,55 @@ export class Visibility {
       this.setState_(state, visible, 0);
     }
 
-    const waitForContinuousTime = config[CONTINUOUS_TIME_MIN]
-        ? config[CONTINUOUS_TIME_MIN] - state[CONTINUOUS_TIME]
-        : Infinity;
-    const waitForTotalTime = config[TOTAL_TIME_MIN]
-        ? config[TOTAL_TIME_MIN] - state[TOTAL_VISIBLE_TIME]
-        : Infinity;
+    listener['state'] = state;
+
+    return this.isConditionMet_(state, config, triggerType);
+  }
+
+  /**
+   * @param {!Object} state
+   * @param {!Object} config
+   * @return {number}
+   * @private
+   */
+  computeTimeToWait_(state, config) {
+    const waitForContinuousTime =
+        config[CONTINUOUS_TIME_MIN] > state[CONTINUOUS_TIME]
```

Comment:
So, config I sort of get, since they are user-authored. But state seems to be completely programmatic and should be typedef'd and used as such.

That being said, ideally I'd typedef config as well and just transform it once and assign all default values and possibly user-error inconsistencies.